kolibri/core/device/serializers.py

```diff
@@ -244,6 +247,26 @@ class Meta:
     def create(self, validated_data):
         raise serializers.ValidationError("Device settings can only be updated")
 
+    def update(self, instance, validated_data):
+        if "extra_settings" in validated_data:
+            extra_settings = validated_data.get("extra_settings")
+            initial_extra_settings = getattr(instance, "extra_settings", None)
```

Comment:
Might want to make the `None` default something that won't fail `json.loads`

Reply:
Reflecting on this some more, I'm not sure the json parsing is even needed. Both values should already be a dictionary. Have you tested this by changing the setting on the Device settings page?